src/pages/CarsList/CarsList.tsx

```diff
@@ -1,66 +1,62 @@
+import { carTypeList } from "../../models/CarType";
+import { getCars } from "../../redux/store/cars/actions";
 import "./CarsList.scss";
-import { Button, ButtonGroup, Container } from "react-bootstrap";
-import { useLocation } from "react-router-dom";
-
 function CarsList() {
   const dispatch = useDispatch();
   const location = useLocation();
-  const [searchtext] = useState<string>(
-    location.state === null ? "" : (location.state as string)
-  );
-
+  const navigate = useNavigate();
+  const queryParams = new URLSearchParams(window.location.search);
+  const [carType, setCarType] = useState<number>(3);
   useEffect(() => {
-    window.scrollTo(0, 0);
-  }, [location]);
-  
-  const [carType, SetCarType] = useState<string>(
-    ["sedan", "SUV", "hatchback", "coupe"].includes(
-      JSON.parse(JSON.stringify(searchtext)).searchtext as string
-    )
-      ? (JSON.parse(JSON.stringify(searchtext)).searchtext as string)
-      : ""
-  );
-  const cars: CarDetails[] = useSelector(
-    (state: RootState) => JSON.parse(JSON.stringify(state.cars)).cars
-  );
-
+    document.title = "Xtreme Cars | All Cars";
+    if (queryParams.get("car-type") === null) navigate("*");
+    let params = queryParams?.get("car-type");
+    switch (params) {
+      case "sedan":
+        setCarType(0);
+        break;
+      case "SUV":
+        setCarType(1);
+        break;
+      case "hatchback":
+        setCarType(2);
+        break;
+      default:
+        setCarType(3);
+        break;
+    }
+  }, []);
   useEffect(() => {
-    dispatch(getCars(carType));
+    if (carType == 3) navigate("/cars");
+    else navigate("/cars?car-type=" + carTypeList[carType]);
+    dispatch(getCars(carTypeList[carType]));
   }, [dispatch, carType]);
-
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [location]);
+  const cars: any = useSelector((state: any) => state.cars.cars);
   return (
     <>
       <ButtonGroup className="mb-2">
-        <Button variant="secondary" onClick={() => SetCarType("")}>
+        <Button onClick={() => setCarType(3)} autoFocus={true}>
           View All
         </Button>
-        <Button variant="secondary" onClick={() => SetCarType("sedan")}>
-          Sedan
-        </Button>
-        <Button variant="secondary" onClick={() => SetCarType("SUV")}>
-          SUV
-        </Button>
-        <Button variant="secondary" onClick={() => SetCarType("hatchback")}>
-          Hatchback
-        </Button>
-        <Button variant="secondary" onClick={() => SetCarType("coupe")}>
-          Coupe
-        </Button>
+        <Button onClick={() => setCarType(0)}>Sedan</Button>
+        <Button onClick={() => setCarType(1)}>SUV</Button>
+        <Button onClick={() => setCarType(2)}>Hatchback</Button>
       </ButtonGroup>
       <Container className="carsContainer">
         {cars?.length === 0 && <div id="nocars">No Cars Available</div>}
-        {cars?.map((car) => (
+        {cars?.map((car: any) => (
```

Comment:
I am pretty sure there is a car detail or something type defined already.
If we had to use any everywhere, why not start using javascript why bother wit Typescript 🤷